configure.ac

```diff
@@ -1306,7 +1306,10 @@ AC_SUBST([jsonsrcdir])
 jsonlib=""
 AC_SUBST([jsonlib])
 
+PAC_PUSH_ALL_FLAGS()
+PAC_RESET_ALL_FLAGS()
 PAC_CONFIG_SUBDIR_ARGS([modules/json-c],[--disable-werror],[],[AC_MSG_ERROR(json-c configure failed)])
+PAC_POP_ALL_FLAGS()
```

Comment:
Just a comment: TODO: we should abstract this and do the same (or similar) to all external packages.

Reply:
We could just add this into `PAC_CONFIG_SUBDIR_ARGS`.  That's meant for external packages anyway.  But we don't need to hold up this PR for that.

Reply:
Was only waiting for the warnings tests... and it seems it glitches up again, have to re-run it.

Reply:
> We could just add this into `PAC_CONFIG_SUBDIR_ARGS`. That's meant for external packages anyway. But we don't need to hold up this PR for that.

We have to differentiate `mpl`. `mpl` is not really external and it is designed to inherit flags from mpich.

Reply:
No, we should modify `mpl` to not rely on mpich flags.

Reply:
In particular, `mpl` are mostly header library, it needs consistent compiler flags for it to work.

Reply:
Disagree.